Sources/HTTPPathCoding/HTTPPathSegment.swift

```diff
@@ -97,10 +104,10 @@ public struct HTTPPathSegment: Equatable {
                 if let variableKey = currentVariableKey {
                     let currentUnparsedValue = try getVariableValue(
                         rawVariableValue: unparsedValue,
-                        remainingSegmentValues: remainingSegmentValues,
+                        remainingSegmentValues: &remainingSegmentValues,
```

Comment:
`getVariableValue()` method name should imply potential consumption of `remainingSegmentValues`